cli/js/compiler.ts

```diff
@@ -1380,21 +1421,94 @@ function compile({
   };
 }
 
+function transpile({
+  config: configText,
+  configPath,
+  cwd,
+  performance,
+  sourceFiles,
+}: TranspileRequest): CompileResponse {
```

Comment:
Nitpick: add distinct `TranspileResponse` type? Otherwise, we might end up adding optional fields just to cater for the fact that both `compile()` and `transpile()` return the same type.